src/wallet/partialTxProposal.ts

```diff
@@ -371,6 +371,29 @@ class PartialTxProposal {
     }
   }
 
+  /**
+   * Overwrites the proposal's signatures with the serialized contents in the parameters
+   * @param serializedSignatures
+   *
+   * @throws {InvalidPartialTxError} Inputs and outputs balance should match before the signatures can be added.
+   */
+  setSignatures(serializedSignatures: string): void {
+    if (!this.partialTx.isComplete()) {
+      // partialTx is not complete, we cannot sign it.
+      throw new InvalidPartialTxError('Cannot sign incomplete data');
+    }
+    const tx: Transaction = this.partialTx.getTx();
+
+    // Creating an empty signatures object
+    this.signatures = new PartialTxInputData(
+      tx.getDataToSign().toString('hex'),
+      tx.inputs.length
+    );
+
+    // Setting the signatures data from the parameters
+    this.signatures.addSignatures(serializedSignatures)
```

Comment:
This method will only work if the signatures are for the same tx (indicated by the `tx.getDataToSign().toString('hex')`).
While this is the expecetd behavior I think we should add a test checking that it fails if we try to add a signature from another tx.

Reply:
Always good to have more validations and improve the error message. It's terrible to have a vague error on these situations.
Added this on [190caaa](https://github.com/HathorNetwork/hathor-wallet-lib/pull/475/commits/190caaa485b7f29dbba783d060570524832eab58)